pkg/cmd/roachtest/registry/operation_spec.go

```diff
@@ -0,0 +1,136 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package registry
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/operation"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/spec"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test"
+)
+
+type OperationDependency int
+
+const (
+	OperationRequiresNodes OperationDependency = iota
+	OperationRequiresSqlConnection
+	OperationRequiresDatabaseSchema
+	OperationRequiresPopulatedDatabase
+)
+
+// OperationSpec is a spec for a roachtest operation.
+type OperationSpec struct {
+	Skip string // if non-empty, operation will be skipped
+	// When Skip is set, this can contain more text to be printed in the logs
+	// after the "--- SKIP" line.
+	SkipDetails string
+
+	Name string
+	// Owner is the name of the team responsible for signing off on failures of
+	// this test that happen in the release process. This must be one of a limited
+	// set of values (the keys in the roachtestTeams map).
+	Owner Owner
+	// The maximum duration the test is allowed to run before it is considered
+	// failed. If not specified, the default timeout is 10m before the test's
+	// associated cluster expires. The timeout is always truncated to 10m before
+	// the test's cluster expires.
```

Comment:
Copy pasted from `TestSpec`, doesn't make much sense here.

Reply:
We definitely need a timeout; updated the comment.